src/realm/object-store/c_api/error.cpp

```diff
@@ -225,7 +225,9 @@ bool ErrorStorage::get_as_realm_error_t(realm_error_t* out) const noexcept
         return false;
     }
 
-    *out = *m_err;
+    if (out) {
```

Comment:
A few tests pass nullptr for out when no error is expected, which made it crash instead of reporting a test failure when an error did happen.